### DISCUSSION TITLE
Modal component: Possibility to prevent close

### DISCUSSION THREAD

**Discussion by chessydk** [tailwindlabs/headlessui#646] [Ideas]
It would be very useful to be able to prevent closing (esc-key / clicking outside). So e.g. a "preventClose" property on Dialog. The dialog would then have to be closed by setting :open to false.

**Comment by livthomas** [comment]
AFAIK, opening/closing the dialog is fully under your control. You can simply ignore `close` event and set `open` prop to whatever you want.

But I would appreciate an ability to ignore clicks on the overlay (configurable via some prop).
